src/common/colors.ts

```diff
@@ -1,53 +1,63 @@
+const secondary = '#FAA21B';
+const gray = '#808080';
+const info = '#0A94C2';
+const success = '#09AA09';
+const warning = '#FFC000';
+const error = '#DF2020';
+
 export const colors = {
-  primaryLighten45: '#73A8CC',
-  primaryLighten30: '#4C90BF',
-  primaryLighten15: '#2678B1',
-  primary: '#0060A3',
-  primaryDarken15: '#00528B',
-  primaryDarken30: '#004372',
-  primaryDarken45: '#00355A',
-
-  secondaryLighten45: '#FCCC82',
-  secondaryLighten30: '#FCBE5F',
-  secondaryLighten15: '#FBB03D',
-  secondary: '#FAA21B',
-  secondaryDarken15: '#D58A17',
-  secondaryDarken30: '#AF7113',
-  secondaryDarken45: '#89590F',
-
-  white: '#FFFFFF',
-  grayLighten90: '#F2F2F2',
-  grayLighten60: '#CCCCCC',
-  grayLighten30: '#A6A6A6',
-  gray: '#808080',
-  grayDarken30: '#5A5A5A',
-  grayDarken60: '#333333',
-  grayDarken90: '#0D0D0D',
-  black: '#000000',
-
-  infoLighten90: '#E6F4F9',
-  infoLighten45: '#78C4DD',
-  info: '#0A94C2',
-  infoDarken35: '#06607E',
-  infoDarken70: '#032C3A',
-
-  successLighten90: '#E6F7E6',
-  successLighten45: '#78D078',
-  success: '#09AA09',
-  successDarken35: '#066E06',
-  successDarken70: '#033303',
-
-  warningLighten90: '#FFF9E5',
-  warningLighten45: '#FFDC73',
-  warning: '#FFC000',
-  warningDarken35: '#A67D00',
-  warningDarken70: '#4D3A00',
-
-  errorLighten90: '#FCE9E9',
-  errorLighten45: '#ED8484',
-  error: '#DF2020',
-  errorDarken35: '#911515',
-  errorDarken70: '#430A0A',
+  primaryLighten45: pSBC(0.45, primary),
+  primaryLighten30: pSBC(0.3, primary),
+  primaryLighten15: pSBC(0.15, primary),
+  primary: primary,
+  primaryDarken15: pSBC(-0.15, primary),
+  primaryDarken30: pSBC(-0.3, primary),
+  primaryDarken45: pSBC(-0.45, primary),
+
+  secondaryLighten45: pSBC(0.45, secondary),
+  secondaryLighten30: pSBC(0.3, secondary),
+  secondaryLighten15: pSBC(0.15, secondary),
+  secondary: secondary,
+  secondaryDarken15: pSBC(-0.15, secondary),
+  secondaryDarken30: pSBC(-0.3, secondary),
+  secondaryDarken45: pSBC(-0.45, secondary),
+
+  white: pSBC(1, gray),
+  grayLighten90: pSBC(0.9, gray),
+  grayLighten60: pSBC(0.6, gray),
+  grayLighten30: pSBC(0.3, gray),
+  gray: gray,
+  grayDarken30: pSBC(-0.3, gray),
+  grayDarken60: pSBC(-0.6, gray),
+  grayDarken90: pSBC(-0.9, gray),
+  black: pSBC(-1, gray),
+
+  infoLighten90: pSBC(0.9, info),
+  infoLighten45: pSBC(0.45, info),
+  info: info,
+  infoDarken35: pSBC(-0.35, info),
+  infoDarken70: pSBC(-0.7, info),
+
+  successLighten90: pSBC(0.9, success),
+  successLighten45: pSBC(0.45, success),
+  success: success,
+  successDarken35: pSBC(-0.35, success),
+  successDarken70: pSBC(-0.7, success),
+
+  warningLighten90: pSBC(0.9, warning),
+  warningLighten45: pSBC(0.45, warning),
+  warning: warning,
+  warningDarken35: pSBC(-0.35, warning),
+  warningDarken70: pSBC(-0.7, warning),
+
+  errorLighten90: pSBC(0.9, error),
+  errorLighten45: pSBC(0.45, error),
+  error: error,
+  errorDarken35: pSBC(-0.35, error),
+  errorDarken70: pSBC(-0.7, error),
```

Comment:
Just some food-for-thought, I don't mind this as-is: do we need to store all these darkened and lightened colors when we have a function to do it for us? I guess it ensures consistency. :man_shrugging:

Reply:
I'm not sure either, I mostly think its useful to have set colors so people have a set amount of shades and tints to take into consideration. If we just have the function, how would we avoid some scrub deciding that `primaryDarken17` is just so much better than `primaryDarken15`?

Reply:
We could restrict the input values to the script to [45, 70, 90, ...] I guess, but yep, it makes sense to restrict the number of shades in use.